Sources/NIOSSL/SSLCallbacks.swift

```diff
@@ -130,6 +130,18 @@ extension KeyLogCallbackManager {
     }
 }
 
+/// PSK Identity Callback Manager
+public typealias NIOPSKIdentityCallback = () -> (key: NIOSSLSecureBytes, identity: NIOSSLSecureBytes)
```

Comment:
I think we need two callbacks, not one. The server is supposed to have access to the identity asserted by the peer so it can choose. Additionally, the client should be given an identity hint. Finally, we should probably let the server be able to configure an identity hint as well.

Reply:
I still have work here, but the callbacks are split now in [7742aa0](https://github.com/apple/swift-nio-ssl/pull/369/commits/7742aa0cc15f7b191af71578fd2227c7041a7cba).

Reply:
I added two different callbacks here on 943aab0. I also tried to address the following: 
1. Send the hint and possibly identity to callback in the`serverPSKCallback`.
2. Configure a hint for the client
3. Configure a hint for server.